### ISSUE TITLE
dpkg error on Debian 11 Stable

### ISSUE THREAD

**Issue by Spirograph12** [chaiNNer-org/chaiNNer#1763]
Thanks for chaiNNer. I've been watching YouTube tutes and it looks like a great program! I'm a home user of Linux, not a coder. I tried installing v0.18.6 .deb version on MX 21.3 (based on Debian 11 Stable). Got this dpkg error:

```
sudo dpkg --install chaiNNer-0.18.6-x64-linux-debian.deb
dpkg-deb: error: archive 'chaiNNer-0.18.6-x64-linux-debian.deb' uses unknown compression for member 'control.tar.zst', giving up
dpkg: error processing archive chaiNNer-0.18.6-x64-linux-debian.deb (--install):
 dpkg-deb --control subprocess returned error exit status 2
Errors were encountered while processing:
 chaiNNer-0.18.6-x64-linux-debian.deb
```
 
This seems to be because dpkg v1.2.0.12 (used by Debian 11 Stable) doesn't support zstd compression:

[StackExchange](https://unix.stackexchange.com/questions/669004/zst-compression-not-supported-by-apt-dpkg)

Forgive me if I have it wrong, but it seems like you might achieve greater Linux compatibility if you built the .deb using older xz compression:

[StackOverflow](https://stackoverflow.com/questions/71196872/how-do-i-build-a-deb-without-zstd-compression-in-ubuntu-21-10)

Though xz may be slower to decompress, is has the advantage of being more universally supported.

Thank you for your time.

**Comment by joeyballentine** [comment]
It doesn't look like i have the option of specifying dpkg version. https://js.electronforge.io/interfaces/_electron_forge_maker_deb.InternalOptions.MakerDebConfigOptions.html

Feel free to open an issue to suggest it to them here: https://github.com/electron/forge

And maybe also suggest an appimage maker as well, because i also cannot package for that.

**Comment by Spirograph12** [comment]
Fair enough. The good news is that the issue may eventually fade as people start adopting Debian 12, which should be officially released later this year.

It looks like you could build a more universal .deb package using [Electron-Builder](https://www.electron.build/generated/linuxtargetspecificoptions), which does seem to let you choose compression type:

LinuxTargetSpecificOptions
...
compression = xz “gz” | “bzip2” | “xz” | “lzo” | “undefined” - The compression type.

BUT... This [short article on beekeeperstudio.io](https://www.beekeeperstudio.io/blog/distribute-electron-apps-for-linux) makes the case that the best way to target the Linux platform is to use Electron-Builder to make an *AppImage* (which is a single binary that will will "just run" on all major Linux distributions). See also this [short article on appimage.org](https://docs.appimage.org/introduction/motivation.html).

AppImages on Linux are similar to PortableApps on Windows. Since an AppImage more or less runs from a single, uninstalled binary, it would tend to skirt the dpkg installation issue we've been discussing, as well as other potential issues with different flavors of Linux.

Where you already have a .deb package, it might be possible to convert it to AppImage format using [pkg2appimage](https://docs.appimage.org/packaging-guide/converting-binary-packages/pkg2appimage.html) or [deb2appimage](https://github.com/simoniz0r/deb2appimage).

chaiNNer is the premier, full-featured, professional app for upscaling and more. But for an example of a very limited upscaling app which nevertheless makes good use of AppImage format, see [here](https://github.com/upscayl/upscayl/releases).

Thanks again for your time. Like I said I'm not a coder, so apologies if some of this is nonsensical.

**Comment by joeyballentine** [comment]
Have you tried the Linux zip? That one should be a portable build

**Comment by Spirograph12** [comment]
I didn't originally see the Linux zip, as my Linux-trained eye passes over anything with what are (traditionally) Windows extensions. It's pretty rare to see a Linux app packaged as .zip

I tried it out, but found no joy. As the attached animated .gif shows, the unzipped folder seems to contain a main file which Debian 11 Stable identifies as a shared library, rather than an executable. Double-clicking on it, the system didn't know how to open or run it.

![Screenshot_chaiNNer_and_Peazip_anim](https://user-images.githubusercontent.com/99771261/235084307-f40f2124-6152-4857-ade1-a9919d55822e.gif)

By contrast (and by way of example), the animated .gif shows the Linux portable of Peazip. This has a file which Debian 11 identifies as an executable. Double-clicking on it opens the Peazip application.

I'm not a Linux evangelist, and don't want to lead you anyplace you don't want to go. I would only say that the popularity of the AppImage format is that it tends to solve many headaches for both developers and end users. Instead of having to tweak releases for all the animals in the Linux distro zoo, developers can concentrate on making *one package* that's fairly universal in its compatibility.

**Comment by mikeisfreeman** [comment]
> I tried it out, but found no joy.

I've had the same experience for the most part, but found a solution, thought I'd share.

In your terminal emulator, go to the location of the unzipped folder, make 'chainner' executable and start it from the terminal.

`chmod +x chainner`

start it like this the first time so you can see output, in case there's more issues.. it can help identify issues by showing you program output.
`./chainner`

after that, if it's working ok, you can create a clickable way to start it easily if you want, perhaps the developers can add this to the zip?
in the chainner directory:
`nano chainner.desktop`

paste this into the file:
```
#!/usr/bin/env xdg-open
[Desktop Entry]
Name=chainner
Comment=chainner
Exec=./chainner
Icon=icon.png
Type=Application
```

and mark it executable:
`chmod +x chainner.desktop`
